src/main/scala/tech/sourced/gemini/Hash.scala

```diff
@@ -33,8 +34,8 @@ class Hash(session: SparkSession, log: Slf4jLogger) {
     * @param repos DataFrame with engine.getRepositories schema
     */
   def forRepos(repos: DataFrame): HashResult = {
-    val files = filesForRepos(repos).cache()
-    val uasts = extractUast(files).cache()
+    val files = filesForRepos(repos).persist(StorageLevel.MEMORY_AND_DISK_SER)
+    val uasts = extractUast(files).persist(StorageLevel.MEMORY_AND_DISK_SER)
```

Comment:
should we make this option configurable?

Reply:
We will, one day, after fine-tuning and understanding performance implications better. For now, it just changes the default persistence level from one to another.